src/main/java/org/cactoos/iterable/IterableOf.java

```diff
@@ -154,12 +159,20 @@ public boolean equals(final Object other) {
                     () -> other != null,
                     () -> Iterable.class.isAssignableFrom(other.getClass()),
                     () -> {
-                        final Iterable<?> compared = (Iterable<?>) other;
-                        final Iterator<?> iterator = compared.iterator();
-                        return new Unchecked<>(
+                        final Iterable<X> compared = (Iterable<X>) other;
+                        return new ScalarWithFallback<>(
                             new And(
-                                (X input) -> input.equals(iterator.next()),
-                                this
+                                (X value) -> new True().value(),
+                                new Matched<>(
+                                    this,
+                                    compared
+                                )
+                            ),
+                            new IterableOf<>(
+                                new FallbackFrom<>(
+                                    IllegalStateException.class,
+                                    ex -> new False().value()
```

Comment:
@iiylll I think you can use directly `true` and `false`, no need to go through an extra layer of indirection

Reply:
@victornoel Yes, I missed that comment. I will fix it.